src/lib/middleware/response-time-metrics.ts

```diff
@@ -2,21 +2,36 @@ import * as responseTime from 'response-time';
 import EventEmitter from 'events';
 import { REQUEST_TIME } from '../metric-events';
 import { IFlagResolver } from '../types/experimental';
+import { InstanceStatsService } from 'lib/services';
+import memoizee from 'memoizee';
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 const _responseTime = responseTime.default;
 
+const countAppNames = async (instanceStatsService: InstanceStatsService) => {
+    const response = await instanceStatsService.getLabeledAppCounts();
+    return response.find((c) => c.range === '7d').count;
+};
 export function responseTimeMetrics(
     eventBus: EventEmitter,
     flagResolver: IFlagResolver,
+    instanceStatsService: InstanceStatsService,
 ): any {
-    return _responseTime((req, res, time) => {
+    const appNameReportingThreshold = 100;
```

Comment:
Maybe add this as a const in `src/lib/util/constants.ts` and import it here?

Reply:
It could be one option, but I was also thinking that we could use variants (we're already using a feature toggle to control the behavior), so we can modify it if needed (close to being a configuration more than a toggle though)

Reply:
Ooh, that's smart!